SignalServiceKit/src/Storage/TSStorageManager.m

```diff
@@ -98,29 +125,18 @@ - (void)runAsyncRegistrationsWithCompletion:(void (^_Nonnull)(void))completion
 {
     OWSAssert(completion);
 
-    // Asynchronously register other extensions.
-    //
-    // All sync registrations must be done before all async registrations,
-    // or the sync registrations will block on the async registrations.
-    [TSDatabaseView asyncRegisterUnseenDatabaseView];
-    [TSDatabaseView asyncRegisterThreadOutgoingMessagesDatabaseView];
-    [TSDatabaseView asyncRegisterThreadSpecialMessagesDatabaseView];
-
-    // Register extensions which aren't essential for rendering threads async.
-    [OWSIncomingMessageFinder asyncRegisterExtensionWithStorageManager:self];
-    [TSDatabaseView asyncRegisterSecondaryDevicesDatabaseView];
-    [OWSDisappearingMessagesFinder asyncRegisterDatabaseExtensions:self];
-    [OWSFailedMessagesJob asyncRegisterDatabaseExtensionsWithStorageManager:self];
-    [OWSFailedAttachmentDownloadsJob asyncRegisterDatabaseExtensionsWithStorageManager:self];
+    runAsyncRegistrationsForPrimaryStorage(self);
 
     // Block until all async registrations are complete.
-    [self.newDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
-        OWSAssert(!self.areAsyncRegistrationsComplete);
+    YapDatabaseConnection *dbConnection = self.newDatabaseConnection;
+    [dbConnection flushTransactionsWithCompletionQueue:dispatch_get_main_queue()
```

Comment:
A small refinement: `flushTransactionsWithCompletionQueue` instead of a NOOP transaction.

Reply:
Cool, TIL.